### PR TITLE
SDK run-test-playbook Command

### PR DESCRIPTION
**Pull request by ChanochShayner**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/40820

## Description
Added a new SDK command `run-test-playbook`.

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

**Review by guykeller — CHANGES_REQUESTED** [review]

File: demisto_sdk/commands/run_test_playbook/test_playbook_runner.py [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+    def get_test_playbook_id(self, file_path):
+        """
+        Get test playbook ID from test playbook file name
+        """
+        test_playbook_data = get_yaml(file_path=file_path)
+        return test_playbook_data.get('id')
+
+    def run_test_playbooks_folder(self, folder_path):
+        """
+        Run all pack test playbooks
```
[the remark this answers is not in the document]
Comment (ChanochShayner) [reply demisto_sdk/commands/run_test_playbook/test_playbook_runner.py:105]: done

File: demisto_sdk/commands/run_test_playbook/test_playbook_runner.py [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+            status_code = self.run_test_playbook_by_id(test_playbook_id)
+
+        return status_code
+
+    def validate_tpb_path(self) -> bool:
+        """
+        Verifies that the input path configuration given by the user is correct
+        :return: The verification result
+        """
+        valid = True
```
[the remark this answers is not in the document]
Comment (ChanochShayner) [reply demisto_sdk/commands/run_test_playbook/test_playbook_runner.py:84]: done

File: demisto_sdk/commands/run_test_playbook/test_playbook_runner.py [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+        if not self.all_test_playbooks:
+            if not self.test_playbook_path:
+                print_color("Error: Missing option '-tpb' / '--test-playbook-path'.", LOG_COLORS.RED)
```
[the remark this answers is not in the document]
Comment (ChanochShayner) [reply demisto_sdk/commands/run_test_playbook/test_playbook_runner.py:87]: done

File: demisto_sdk/commands/run_test_playbook/test_playbook_runner.py [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+        """
+        status_code = SUCCESS_RETURN_CODE
+        all_tpb_files_list: list = []
+
+        if not self.validate_tpb_path():
+            status_code = ERROR_RETURN_CODE
+
+        # Run all repo test playbooks
+        elif self.all_test_playbooks:
+            all_tpb_files_list.extend(self.run_all_test_playbooks())
```
[the remark this answers is not in the document]
Comment (ChanochShayner) [reply demisto_sdk/commands/run_test_playbook/test_playbook_runner.py:62]: done

File: demisto_sdk/commands/run_test_playbook/test_playbook_runner.py [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+        exit_code: int = self.manage_and_run_test_playbooks()
+        return exit_code
+
+    def manage_and_run_test_playbooks(self):
+        """
+        Manages all ru-test-playbook command flows
+        return The exit code of each flow
+        """
+        status_code = SUCCESS_RETURN_CODE
+        all_tpb_files_list: list = []
```
[the remark this answers is not in the document]
Comment (ChanochShayner) [reply demisto_sdk/commands/run_test_playbook/test_playbook_runner.py:55]: done

File: demisto_sdk/commands/run_test_playbook/test_playbook_runner.py [hunk 6]
```diff
@@ -0,0 +1,230 @@
...
+        self.all_test_playbooks = all
+        self.should_wait = wait
+        self.timeout = timeout
+
+        # we set to None so demisto_client will use env var DEMISTO_VERIFY_SSL
+        self.verify = (not insecure) if insecure else None
+        self.demisto_client = demisto_client.configure(verify_ssl=self.verify)
+        self.base_link_to_workplan = self.get_base_link_to_workplan()
+
+    def run_test_playbooks(self) -> int:
```
[the remark this answers is not in the document]
Comment (ChanochShayner) [reply demisto_sdk/commands/run_test_playbook/test_playbook_runner.py:41]: done

File: demisto_sdk/commands/run_test_playbook/test_playbook_runner.py [hunk 7]
```diff
@@ -0,0 +1,230 @@
...
+        elif self.all_test_playbooks:
+            all_tpb_files_list.extend(self.run_all_test_playbooks())
+
+        # Run all pack test playbooks
+        elif os.path.isdir(self.test_playbook_path):
+            all_tpb_files_list.extend(self.run_test_playbooks_folder(self.test_playbook_path))
+
+        # Run specific test playbook
+        elif os.path.isfile(self.test_playbook_path):
+            all_tpb_files_list.append(self.test_playbook_path)
```
[the remark this answers is not in the document]
Comment (ChanochShayner) [reply demisto_sdk/commands/run_test_playbook/test_playbook_runner.py:70]: done

File: demisto_sdk/commands/run_test_playbook/test_playbook_runner.py [hunk 8]
```diff
@@ -0,0 +1,230 @@
...
+        Run all the repo test playbooks
+        """
+        tpb_list: list = []
+        packs_list = os.listdir('Packs')
+        for pack in packs_list:
+            if os.path.isdir(f'Packs/{pack}'):
+                tpb_list.extend(self.run_test_playbooks_folder(f'Packs/{pack}'))
+        return tpb_list
+
+    def run_test_playbook_by_id(self, test_playbook_id):
```
[the remark this answers is not in the document]
Comment (ChanochShayner) [reply demisto_sdk/commands/run_test_playbook/test_playbook_runner.py:123]: done

File: demisto_sdk/commands/run_test_playbook/test_playbook_runner.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (ChanochShayner) [reply demisto_sdk/commands/run_test_playbook/test_playbook_runner.py:70]: done

File: demisto_sdk/commands/run_test_playbook/test_playbook_runner.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (ChanochShayner) [reply demisto_sdk/commands/run_test_playbook/test_playbook_runner.py:41]: removed

**Review by guykeller — APPROVED** [review]

**Pull request merged**